### PR TITLE
devops: compose up from mirror - no builds to deploy

### PR DESCRIPTION
**Pull request by tfolbrecht**
This PR modified the docker-compose.yaml to allow users to deploy the app without building.
It adds dockerfiles to pin out mirrored versions and custom entrypoints to simplify configuration.
It also assumes there's a top level .env with all necessary variables

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,300 +1,198 @@
...
-      - AWS_REGION=${AWS_REGION}
-      - SPARSE_SERVER_ORIGIN=${SPARSE_SERVER_ORIGIN}
-      - EMBEDDING_SERVER_ORIGIN=${EMBEDDING_SERVER_ORIGIN}
-      - RERANKER_SERVER_ORIGIN=${RERANKER_SERVER_ORIGIN}
-      - UNLIMITED=${UNLIMITED}
-      - REDIS_CONNECTIONS=${REDIS_CONNECTIONS}
+      - HF_TOKEN=${HF_TOKEN}
+      - API_KEY=${OPENAI_API_KEY}
+    env_file:
+      .env
```
Comment (tfolbrecht) [inline docker-compose.yml:111]: The splade embeddings and ingestion-service should be expanded out instead of reading .env

Comment (fastfists) [reply docker-compose.yml:111]: Ingestion-service can be the exact same as the server, just annoying to have to double the size of the docker-compose.yml file

**Comment by fastfists** [comment]
Can you rebase and squash a few of the badly named commits.

File: docker-compose.yml (hunk 1 above)
Comment (fastfists) [reply docker-compose.yml:111]: Splade-embeddings only needs 
```
HF_TOKEN
API_KEY
and SENTRY_URL
```

**Comment by fastfists** [comment]
Also do we want to just have a single large .env.example file instead of having 3-4 different ones at this point. The example .env seems small and reasonable enough to just have comments of what each variable is for, could simply users lives.

**Review by fastfists — CHANGES_REQUESTED** [review]
Seems to not be able to connect to keycloak. Set for re-review once it gets set up.

**Comment by skeptrunedev** [comment]
Closing for now pending getting Sentry up

**Pull request closed without merging**